Project configuration

junkerm opened the pull request:
Enables configuring projects with connectors and exporters. 
Connectors can authenticate user credentials

maximilian-junker committed:
first working version of project configuration

maximilian-junker committed:
wip

junkerm committed:
Merge remote-tracking branch 'origin/develop' into project-configuration

Conflicts:
	bundles/specmate-config/config/specmate-config.properties

junkerm committed:
wip: configure exporter

junkerm committed:
wip: first working version of auth against hp proxy

junkerm committed:
adding some comments

munterkalmsteiner reviewed (approved):
Looks good to me (except for the integration-test failures).

junkerm committed:
Merge remote-tracking branch 'origin/develop' into project-configuration

Conflicts:
	bundles/specmate-auth/bnd.bnd
	bundles/specmate-auth/src/com/specmate/auth/internal/AuthenticationServiceImpl.java
	bundles/specmate-config/config/specmate-config.properties
	web/src/app/modules/data/modules/data-service/services/service-interface.ts

munterkalmsteiner committed:
Missing bundles

munterkalmsteiner committed:
Cater for incomplete config

munterkalmsteiner committed:
Don't load projects that are not completely configured

munterkalmsteiner committed:
Merge branch 'develop' into project-configuration

munterkalmsteiner committed:
Removing again config service from integration tests

If we use the config service here, we cannot manually configure the services
as needed for testing.

munterkalmsteiner committed:
Project service that does not pull in the config service

munterkalmsteiner committed:
Better return a new project instead of null

munterkalmsteiner commented:
I have now found the root cause of the problem. The project service needs the config service to startup. The authentication service, in turn, needs the project service. The integration tests need the authentication service. However, the integration tests need to be configured manually, and when we startup the config service (by proxy through authentication and project service), we loose that control. 

I've solved that by implementing a dummy project service in the integration test bundle that doesn't pull in the config service.

Commit:
Projects as services

- make projects separate services
- wire connectors and exporters to projects by osgi ds

junkerm commented:
I refactored the way projects and connectors and exporters are wired together. Projects are now separate services that can be created independent from the configuration (e.g. for the dummy data).

munterkalmsteiner commented on bundles/specmate-integration-test/src/com/specmate/test/integration/support/DummyProjectService.java:
```diff
@@ -13,8 +15,28 @@
+			public IExportService getExporter() {
+				// TODO Auto-generated method stub
+				return null;
+			}
```
Remove TODOs

munterkalmsteiner reviewed (approved):
Just a minor comment. Good solution! Ready to merge.

junkerm committed:
remove todos

junkerm committed:
Merge remote-tracking branch 'origin/develop' into project-configuration

Conflicts:
	bundles/specmate-auth/src/com/specmate/auth/internal/AuthenticationServiceImpl.java
	bundles/specmate-dummy-data/src/com/specmate/dummydata/DummyDataService.java

junkerm committed:
update run-file for auth tests

munterkalmsteiner committed:
Use DummyProject and service in Authentication tests
